packages/grafana-ui/src/components/Segment/Segment.story.tsx

```diff
@@ -99,7 +103,7 @@ export const GroupedArrayOptions = () => {
 };
 
 export const CustomOptionsAllowed = () => {
-  const [value, setValue] = useState(options[0]);
+  const [value, setValue] = useState<SelectableValue>(options[0]);
```

Comment:
```suggestion
  const [value, setValue] = useState<SelectableValue<string | number>>(options[0]);
```

you'll also need to remove the destructuring in the `onChange`, e.g.
```tsx
        onChange={(value) => {
          setValue(value);
          action('Segment value changed')(value);
        }}
```